mllib/src/main/scala/org/apache/spark/mllib/feature/ChiSqSelector.scala

```diff
@@ -57,22 +69,21 @@ class ChiSqSelectorModel @Since("1.3.0") (
    * Preserves the order of filtered features the same as their indices are stored.
    * Might be moved to Vector as .slice
    * @param features vector
-   * @param filterIndices indices of features to filter
+   * @param filterIndices indices of features to filter, must be ordered asc
    */
   private def compress(features: Vector, filterIndices: Array[Int]): Vector = {
-    val orderedIndices = filterIndices.sorted
```

Comment:
@yanboliang there's nothing urgent about this or that change. It was not appropriate to merge this. This change undoes an improvement on an existing change several of us reviewed for weeks and isn't actually necessary. I'm going to revert and propose a change that makes less change but still enables the improvement you wanted to make.